tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/sdk/BallerinaSdkUtils.java

```diff
@@ -320,6 +320,34 @@ public static VirtualFile getSdkSrcDir(@NotNull Project project, @Nullable Modul
         });
     }
 
+    /**
+     * Searches for a ballerina project using outward recursion starting from the file directory, until the given root
+     * directory is found. Returns and empty string if unable to detect any ballerina project under the current intellij
+     * project source root.
+     */
+    public static String searchForBallerinaProjectRoot(String currentPath, String root) {
+
+        if (currentPath.equals(root) || currentPath.equals("") || root.equals("")) {
```

Comment:
I think we can use `isEmpty()` instead of `equals()`.

Reply:
Fixed :)